src/Cli/dotnet/NuGetSignatureVerificationEnabler.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Tools.MSBuild;
+
+namespace Microsoft.DotNet.Tools
+{
+    public static class NuGetSignatureVerificationEnabler
+    {
+        private static readonly EnvironmentProvider s_environmentProvider = new();
+
+        internal static readonly string DotNetNuGetSignatureVerification = "DOTNET_NUGET_SIGNATURE_VERIFICATION";
+
+        public static void ConditionallyEnable(ForwardingApp forwardingApp, IEnvironmentProvider? environmentProvider = null)
```

Comment:
Is it possible to extract the logic from these two overloaded methods, which are identical in their implementation, and create a new method to contain that logic, which can then be called by both of the overloaded methods?

Reply:
Their implementation is similar but not identical.  Crucially, how you set environment variables for `ForwardingApp` and `MSBuildFowardingApp` instances is different.  See [here](https://github.com/dotnet/sdk/pull/31868/files#diff-8777ab94208f467bed9672d3f15e8d1343514f51dc1771f2049c9cbc7a02e2ffR29) and [here](https://github.com/dotnet/sdk/pull/31868/files#diff-8777ab94208f467bed9672d3f15e8d1343514f51dc1771f2049c9cbc7a02e2ffR45).